app/values/maybe_uri.rb

```diff
@@ -0,0 +1,21 @@
+class MaybeURI
+  pattr_initialize :raw_value
+
+  def value
+    if is_uri?
+      uri
+    else
+      raw_value
+    end
+  end
+
+  private
+
+  def is_uri?
```

Comment:
I'm doing something similar https://github.com/OregonDigital/oregondigital_2/pull/107/files#diff-e48609f9c951b5c780bdc7c84335d30aR22 here. Can we do URICheck.new(raw_value).uri? Or add a "uri?" method here, I'm ambivalent.